ocs_ci/ocs/platform_nodes.py

```diff
@@ -220,6 +223,31 @@ def get_vms(self, nodes):
             vms.extend(node_vms)
         return vms
 
+    def get_vm_nodes_in_dc(self, nodes, dc=None):
+        """
+        Get vSphere vm objects list in the Datacenter(and not just in the cluster scope)
+
+        Args:
+            nodes (list): The OCS objects of the nodes
+            dc (str): The Datacenter name. If not specified, it will use the current dc.
+
+        Returns:
+            list: vSphere vm objects list in the Datacenter
+
+        """
+        vms_in_dc = self.vsphere.get_all_vms_in_dc(self.datacenter)
+        node_names = set([node.get().get("metadata").get("name") for node in nodes])
+        vms = []
+        for vm in vms_in_dc:
+            try:
+                vm_name = vm.name
+                if vm_name in node_names:
+                    vms.append(vm)
+            except Exception as e:
```

Comment:
What is the expected exception here ? ```vms``` will be incomplete list of VMs in case of an Exception. Please add a validation before returning if ```len(nodes)``` should match ```len(vms)```

Reply:
There is a chance that one of the VM objects will not exist, so it will throw an exception when trying to get the VM name.

Reply:
Resolved after a discussion.